downloader/src/main/kotlin/VcsHost.kt

```diff
@@ -307,6 +338,19 @@ enum class VcsHost(
             return values().find { host -> host.isApplicable(vcsInfo) }
                 ?.toPermalinkInternal(vcsInfo.normalize(), startLine, endLine)
         }
+
+        /**
+         * Returns the [VcsHost] for a [vcsUrl].
+         */
+        fun fromUri(vcsUrl: URI): VcsHost? {
+            return when {
+                vcsUrl.host.contains(GITHUB.hostname) -> GITHUB
+                vcsUrl.host.contains(GITLAB.hostname) -> GITLAB
+                vcsUrl.host.contains(BITBUCKET.hostname) -> BITBUCKET
+                vcsUrl.host.contains(SOURCEHUT.hostname) -> SOURCEHUT
+                else -> null
```

Comment:
IMO we should write this code in such a way so we don't need to maintain it when new VCS host providers are added. We can do so by looping over the `enum` values very similar to like we do at

https://github.com/oss-review-toolkit/ort/blob/621b15fb59a006c54755340680edcf7617f01db6/downloader/src/main/kotlin/VcsHost.kt#L245-L251

In fact, I believe the first step (in a separate commit) should be to factor out the required functionality from above's code a `toVcsHost()` function (instead of this one, and use `toVcsHost()` inside `toVcsInfo()`.